pkgs/tools/security/fprintd/default.nix

```
@@ -1,42 +1,66 @@
-{ thinkpad ? false
-, stdenv
-, fetchurl
+{ stdenv
+, fetchFromGitLab
 , fetchpatch
 , pkgconfig
+, meson
+, ninja
 , intltool
-, libfprint-thinkpad ? null
-, libfprint ? null
+, libpam-wrapper
+, cairo
+, gtk-doc
+, libxslt
+, docbook_xsl
 , glib
 , dbus-glib
 , polkit
 , nss
 , pam
 , systemd
-, autoreconfHook
-, gtk-doc
+, libfprint
+, python37Packages
 }:
 
 stdenv.mkDerivation rec {
-  pname = "fprintd" + stdenv.lib.optionalString thinkpad "-thinkpad";
-  version = "0.9.0";
+  pname = "fprintd";
+  version = "1.90.1";
 
-  src = fetchurl {
-    url = "https://gitlab.freedesktop.org/libfprint/fprintd/uploads/9dec4b63d1f00e637070be1477ce63c0/fprintd-${version}.tar.xz";
-    sha256 = "182gcnwb6zjwmk0dn562rjmpbk7ac7dhipbfdhfic2sn1jzis49p";
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "libfprint";
+    repo = pname;
+    rev = version;
+    sha256 = "0mbzk263x7f58i9cxhs44mrngs7zw5wkm62j5r6xlcidhmfn03cg";
   };
 
+  outputs = [ "out" "devdoc" ];
+
   patches = [
+    # Fixes issue with ":" used when multiple paths (might be the case on NixOS)
+    # https://gitlab.freedesktop.org/libfprint/fprintd/-/merge_requests/50
     (fetchpatch {
-      url = "https://gitlab.freedesktop.org/libfprint/fprintd/merge_requests/16.patch";
-      sha256 = "1y39zsmxjll9hip8464qwhq5qg06c13pnafyafgxdph75lvhdll7";
+      url = "https://gitlab.freedesktop.org/libfprint/fprintd/-/merge_requests/50.patch";
+      sha256 = "1wqkk7bqig0fhjbq1487kxkh7yc9356qjsac1x27pzgakbayxpyy";
     })
   ];
 
+  pythonPath = with python37Packages; [
+    python-dbusmock
+    dbus-python
+    pygobject3
+    pycairo
+    libpam-wrapper
+  ];
+
   nativeBuildInputs = [
-    intltool
     pkgconfig
-    autoreconfHook # Drop with above patch
-    gtk-doc # Drop with above patch
+    meson
+    ninja
+    intltool
```

```suggestion
    gettext
```
https://gitlab.freedesktop.org/libfprint/fprintd/-/commit/53d80c14742b8dd387aa449360e2b68bc0291e64

---

I've tried to remove `intltool` to put `gettext` only, but it fails with the following error:

```
meson.build:84:0: ERROR: Program(s) ['pod2man'] not found or not executable
```

---

then you need to add `perl # for pod2man` as well.

---

That was it, thank you.